toolkit/components/uniffi-bindgen-gecko-js/src/templates/js/Object.jsm

```diff
@@ -33,9 +35,20 @@ class {{ object.nm() }} {
 
     {%- for meth in object.methods() %}
     {{ meth.nm() }}({{ meth.arg_names() }}) {
-        {% call js::call_method(meth, type_) %}
+        {% call js::call_method(meth, type_, object) %}
     }
     {%- endfor %}
+
+    destroy() {
+        this.destroyed = true;
+        if (this.callCounter === 0) {
+            return {{ ci.scaffolding_name() }}.{{ object.ffi_object_free().nm() }}(this.ptr);
+        }
+        // This means there are ongoing calls that haven't concluded
+        // yet. The function calls themselves will make sure to deallocate the object once the last
+        // one concludes and we will prevent any new calls by throwing a UniFFIError
+        return Promise.resolve();
```

Comment:
What do you think about making `destroy()` not return anything?  I don't see how a caller can use this return value.